rimapi/helpers.py

```diff
@@ -0,0 +1,17 @@
+from hashlib import sha256
+from django.conf import settings
+
+API_SECRET = getattr(settings, "API_SECRET", 'test_secret_key')
+
+def get_client_ip(request):
+    x_forwarded_for = request.META.get('HTTP_X_FORWARDED_FOR')
+    if x_forwarded_for:
+        ip = x_forwarded_for.split(',')[0]
+    else:
+        ip = request.META.get('REMOTE_ADDR')
+    return ip
+
+def validate_req_hash(data, key):
+    if all (k in data for k in ('hash', key)) and not ('' == data['hash'] == data[key]) and data['hash'].lower() == sha256((str(data[key]) + API_SECRET).encode('utf-8')).hexdigest():
```

Comment:
`not ('' == data['hash'] == data[key])`
expands out to:
`not ('' == data['hash'] and data['hash'] == data[key])`

But don't we want that to be something more like:
`not ('' == data['hash'] or '' == data[key])`?

I could be interpreting this wrong. For some reason I'm having a hard time wrapping my head around this one.

Reply:
I was looking at different ways to handle these in a dict to check for multiple keys, I guess I misinterpreted how it works, It does translate to what you specified. Changed it to .get() now which works to check if the key exists and also validate an empty string as its a false value in python.